llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtBase.td

```diff
@@ -44,62 +44,60 @@ def AnyRankedTensorOrMemRefType : RankedTensorOrMemRefOf<[AnyType]>;
-    : I32EnumAttrCase<"MATMUL_F32F32F32_RHS", 1>;
-def MATMUL_F32F32F32_RESULT
-    : I32EnumAttrCase<"MATMUL_F32F32F32_RESULT", 2>;
-def MATMUL_I8I8I32_LHS
-    : I32EnumAttrCase<"MATMUL_I8I8I32_LHS", 3>;
-def MATMUL_I8I8I32_RHS
-    : I32EnumAttrCase<"MATMUL_I8I8I32_RHS", 4>;
-def MATMUL_I8I8I32_RESULT
-    : I32EnumAttrCase<"MATMUL_I8I8I32_RESULT", 5>;
-def MATMUL_F16F16F32_LHS
-    : I32EnumAttrCase<"MATMUL_F16F16F32_LHS", 6>;
-def MATMUL_F16F16F32_RHS
-    : I32EnumAttrCase<"MATMUL_F16F16F32_RHS", 7>;
-def MATMUL_F16F16F32_RESULT
-    : I32EnumAttrCase<"MATMUL_F16F16F32_RESULT", 8>;
-def MATMUL_F16F16F16_LHS
-    : I32EnumAttrCase<"MATMUL_F16F16F16_LHS", 9>;
-def MATMUL_F16F16F16_RHS
-    : I32EnumAttrCase<"MATMUL_F16F16F16_RHS", 10>;
-def MATMUL_F16F16F16_RESULT
-    : I32EnumAttrCase<"MATMUL_F16F16F16_RESULT", 11>;
-def MATMUL_BF16BF16F32_LHS
-    : I32EnumAttrCase<"MATMUL_BF16BF16F32_LHS", 12>;
-def MATMUL_BF16BF16F32_RHS
-    : I32EnumAttrCase<"MATMUL_BF16BF16F32_RHS", 13>;
-def MATMUL_BF16BF16F32_RESULT
-    : I32EnumAttrCase<"MATMUL_BF16BF16F32_RESULT", 14>;
-def MATMUL_BF16BF16BF16_LHS
-    : I32EnumAttrCase<"MATMUL_BF16BF16BF16_LHS", 15>;
-def MATMUL_BF16BF16BF16_RHS
-    : I32EnumAttrCase<"MATMUL_BF16BF16BF16_RHS", 16>;
-def MATMUL_BF16BF16BF16_RESULT
-    : I32EnumAttrCase<"MATMUL_BF16BF16BF16_RESULT", 17>;
-
-def TensorEncodingEnum
-    : I32EnumAttr<"TensorEncoding",
-                  "identifier for encoding used for the tensor",[
-                    MATMUL_F32F32F32_LHS, MATMUL_F32F32F32_RHS, MATMUL_F32F32F32_RESULT,
-                    MATMUL_I8I8I32_LHS, MATMUL_I8I8I32_RHS, MATMUL_I8I8I32_RESULT,
-                    MATMUL_F16F16F32_LHS, MATMUL_F16F16F32_RHS, MATMUL_F16F16F32_RESULT,
-                    MATMUL_F16F16F16_LHS, MATMUL_F16F16F16_RHS, MATMUL_F16F16F16_RESULT,
-                    MATMUL_BF16BF16F32_LHS, MATMUL_BF16BF16F32_RHS, MATMUL_BF16BF16F32_RESULT,
-                    MATMUL_BF16BF16BF16_LHS, MATMUL_BF16BF16BF16_RHS, MATMUL_BF16BF16BF16_RESULT,
-                  ]> {
+class IREELinalgExt_I32EnumAttr<string name, string summary, list<I32EnumAttrCase> cases>
+    : I32EnumAttr<name, summary, cases> {
   let cppNamespace = "::mlir::iree_compiler::IREE::LinalgExt";
   let genSpecializedAttr = 0;
 }
 
-def TensorEncodingAttr :
-  EnumAttr<IREELinalgExt_Dialect, TensorEncodingEnum, "encoding"> {
-  let summary = [{tensor layout encoding}];
-  let description = [{
+class IREELinalgExt_EnumAttr<EnumAttrInfo enumInfo, string name = "">
+  : EnumAttr<IREELinalgExt_Dialect, enumInfo, name>;
+
+def MATMUL_F32F32F32 : I32EnumAttrCase<"MATMUL_F32F32F32", 0>;
+def MATMUL_I8I8I32 : I32EnumAttrCase<"MATMUL_I8I8I32", 1>;
+def MATMUL_F16F16F32 : I32EnumAttrCase<"MATMUL_F16F16F32", 2>;
+def MATMUL_F16F16F16 : I32EnumAttrCase<"MATMUL_F16F16F16", 3>;
+def MATMUL_BF16BF16F32 : I32EnumAttrCase<"MATMUL_BF16BF16F32", 4>;
+def MATMUL_BF16BF16BF16 : I32EnumAttrCase<"MATMUL_BF16BF16BF16", 5>;
+
+def EncodingOpKind
```

Comment:
I have updated this PR with a tentative renaming along these lines, as a separate commit so it's easy to change into anything else, particularly as it was generated by a command line:

```
git grep --name-only -e '\(\bop_kind\b\|[oO]pKind\b\)' | grep -v dispatch_profiler | xargs sed -i 's/\bop_kind\b/user/g;s/opKind/user/g;s/OpKind/User/g'
```

So this is doing `User` not `UserKind` but we can do that as well. See my comment at https://discord.com/channels/689900678990135345/690274711523164166/1128034873718878290

Reply:
Note: avoiding the `Info` terminology because that is already how we refer to the _result_ of MaterializeEncoding (`MaterializeEncodingInfo` is a struct holding materialized tile sizes).